docs/analyzer/recognizer_registry_provider.md

```diff
@@ -0,0 +1,92 @@
+# Customizing recognizer registry
+To load recognizers from file, use `RecognizerRegistryProvider` to instantiate the recognizer registry and then pass it through to the analyzer engine:
+
+```python
+from presidio_analyzer import AnalyzerEngine
+from presidio_analyzer.recognizer_registry import RecognizerRegistryProvider
+
+recognizer_registry_conf_file = "./analyzer/recognizers-config.yml"
+
+provider = RecognizerRegistryProvider(
+                conf_file=recognizer_registry_conf_file
+            )
+registry = provider.create_recognizer_registry()
+analyzer = AnalyzerEngine(
+            registry=registry,
+            )
+
+results = analyzer.analyze(text="My name is Morris")
+print(results)
+```
+
+## Configuration file structure
+
+```yaml
+global_regex_flags: 26
+recognizers: 
+...
+```
+
+The configuration file consists of two parts:
+
+  - `global_regex_flags`: regex flags to be used in regex matching.
+  - `recognizers`: a list of recognizers to be loaded by the recognizer registry. This list consists of two different types of recognizers: 
+    - Predefined: A set of already defined recognizer classes in presidio
+    - Custom: custom created pattern recognizers that are created ad hoc based on the fields provided in the configuration file
```

Comment:
Ad-hoc recognizer is a different thing in Presidio- a recognizer you can add to the request without having to register it in the registry. I would suggest using a different terminology here to avoid confusion.

Reply:
Better to just discard it then